### PR TITLE
Moodbased coins

### LINKED ISSUE #122: Drop coins based on mood
**Description**
Vikings should wait a bit before dropping coins

**Subtasks**
- [x] Vikings drop coins with a short delay when fulfilling a desire
- [x] Vikings drop coins when all their desires are fulfilled.
- [x] The value/amount of coins dropped is based on mood

**Additional context**
We want to test both the first and second task to see which one we like the most. Maybe use both??

### PR DESCRIPTION
**Pull request by MrJackey**
Closes #122 

**Description**  
A description of the changes made.

**List of changes**  
- Added math utility class
- Created a new state which handles dropping coins
- The amount of coins is now based on the vikings mood
- The vikings drop extra coins when leaving

**Additional context**  
I was unable to find a fitting name for "MoodWhenDesireFulfilled", let me know if you have better suggestion.

image

### CODE REVIEW

File: Assets/00_Content/Scripts/Vikings/States/SatisfiedVikingState.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+			return nextState;
+		}
+
+		private void SetupDroppingCoins() {
+			if (viking.CurrentDesireIndex < viking.Desires.Length) {
+				coinsToDrop = CalculateCoinsToDrop(viking.Stats.Mood);
+			}
+			else {
+				float avgMood = viking.MoodWhenDesireFulfilled.Sum(x => x) /
```
Comment (DreamingJohnny) [inline Assets/00_Content/Scripts/Vikings/States/SatisfiedVikingState.cs:45]: This variable name... proud of it? ;-)

Comment (DreamingJohnny) [reply Assets/00_Content/Scripts/Vikings/States/SatisfiedVikingState.cs:45]: Please explain this calculation, I'm sure it is right, but I would appreciate understanding it a bit more.

File: Assets/00_Content/Scripts/Vikings/Viking.cs [hunk 2]
```diff
@@ -22,6 +23,7 @@ public class Viking : Interactable {
 
 		public VikingData Data => vikingData;
 		public DesireData[] Desires => vikingData.desires;
+		public List<float> MoodWhenDesireFulfilled { get; } = new List<float>();
```
Comment (DreamingJohnny) [inline Assets/00_Content/Scripts/Vikings/Viking.cs:26]: "MoodWhenDone"?
"MoodLeaving"?
"FinalMood"?
"MoodResult"?

**Review by DreamingJohnny — APPROVED** [review]

File: Assets/00_Content/Scripts/Vikings/States/SatisfiedVikingState.cs (hunk 1 above)
Comment (MrJackey) [reply Assets/00_Content/Scripts/Vikings/States/SatisfiedVikingState.cs:45]: I'm calcuating the average mood the viking had when a desire is fulfilled. I'm doing this by adding together all values and dividing the sum with the amount of values.

**Pull request merged**